docs/dev/design/new-search-api.rst

```diff
@@ -0,0 +1,297 @@
+             "highlights": {
+               "title": [
+                 "Full-<span>Text</span> Search"
+               ],
+               "content": []
+             }
+           },
+           {
+             "type": "domain",
+             "role": "http:post",
+             "name": "/api/v3/projects/",
+             "id": "post--api-v3-projects-",
+             "content": "Import a project under authenticated user. Example request: BashPython$ curl \\ -X POST \\ -H \"Authorization: Token <token>\" https://readthedocs.org/api/v3/projects/ \\ -H \"Content-Type: application/json\" \\ -d @body.json import requests import json URL = 'https://readthedocs.org/api/v3/projects/' TOKEN = '<token>' HEADERS = {'Authorization': f'token {TOKEN}'} data = json.load(open('body.json', 'rb')) response = requests.post( URL, json=data, headers=HEADERS, ) print(response.json()) The content of body.json is like, { \"name\": \"Test Project\", \"repository\": { \"url\": \"https://github.com/readthedocs/template\", \"type\": \"git\" }, \"homepage\": \"http://template.readthedocs.io/\", \"programming_language\": \"py\", \"language\": \"es\" } Example response: See Project details Note Read the Docs for Business, also accepts",
+             "highlights": {
+               "name": [],
+               "content": [
+                 ", json=data, headers=HEADERS, ) print(response.json()) The content of body.json is like,  &quot;name&quot;: &quot;<span>Test</span>"
+               ]
+             }
+           }
+         ]
+       }
+     ]
+   }
+
+Examples
+--------
+
+- ``project:docs project:dev/latest test``: search for ``test`` in the default
+  version of the ``docs`` project, and in the latest version of the ``dev`` project.
+- ``a project:docs/stable search term``: search for ``a search term`` in the
+  stable version of the ``docs`` project.
+
+- ``project:docs project\:project/version``: search for ``project::project/version`` in the
+  default version of the ``docs`` project.
+
+- ``search``: invalid, at least one project is required.
+
+Dashboard search
+----------------
+
+This is the search feature that you can access from
+the readthedocs.org/readthedocs.com domains.
+
+We have two types:
+
+Project scoped search:
+   Search files and versions of the curent project only.
+
+Global search:
+   Search files and versions of all projects in .org,
+   and only the projects the user has access to in .com.
+
+   Global search also allows to search projects by name/description.
+
+This search also allows you to see the number of results
+from other projects/versions/sphinx domains (facets).
+
+Project scoped search
+~~~~~~~~~~~~~~~~~~~~~
+
+Here the new syntax won't have effect,
+since we are searching for the files of one project only!
+
+Another approach could be linking to the global search
+with ``project:{project.slug}`` filled in the query.
```

Comment:
If we do this, we could do something similar to github, having a global search bar, and offering the scopes depending if you are on a project/organization or not.

![Screenshot from 2022-08-31 19-02-43](https://user-images.githubusercontent.com/4975310/187805553-82067f50-c9be-41cb-bad9-05c10d9a271f.png)

Reply:
Yea, that is what I'm imagining. Basically we just prepend the project slug in project search mode, but the same UX can be used globally.

Reply:
So, are we okay for now to just redirect to the global search with the filter set when clicking on `Search` in

![Screenshot 2022-09-01 at 12-50-19 docs Read the Docs](https://user-images.githubusercontent.com/4975310/187979994-ec682639-6413-4a98-962a-333461d2e5b9.png)

or do we want to have the new search bar together with this implementation?